src/utilities/urlAssembly.ts

```diff
@@ -1,8 +1,9 @@
 import { Param, ResolvedRoute } from '@/types'
+import { mergeParams } from '@/utilities/mergeParams'
 import { setParamValuesOnUrl } from '@/utilities/paramsFinder'
 
 export function assembleUrl(route: ResolvedRoute, values: Record<string, unknown[]> = {}): string {
-  const params = Object.entries<Param[]>(route.params)
+  const params = Object.entries<Param[]>(mergeParams(route.pathParams, route.queryParams))
```

Comment:
this should work fine, but given that these are now separate on the resolved route, I probably would have assigned values when building the url separately. Seems silly to merge together but required less refactoring